### PR TITLE
Issue #361 - Support for php:// stream wrappers in \PDepend\Engine

### LINKED ISSUE #361: Support for php:// stream wrappers in \PDepend\Engine
The problem is that the `\PDepend\Engine::createFileIterator()` iterates over the `$this->files` which contains a file path like this `php://stdin`, but that file path is added to the `$fileIterator` like this:
`$fileIterator->append(new Iterator(new \GlobIterator($file), $this->fileFilter));`,
and later the `\PDepend\Input\Iterator::getFullPath()` is called, which calls the `::getRealPath()` method of the current item, which is an instance of `\SplFileObject`, and it always returns false.

In short this is the problem:
```php
$item = new \SplFileObject('php://stdin');
$path = $item->getRealPath(); // $path === false
```

Original issue: [phpmd/phpmd#542]

### PR DESCRIPTION
**Pull request by Sweetchuck**
Fix #361
Fix https://github.com/phpmd/phpmd/issues/542

### CODE REVIEW

**Comment by emirb** [comment]
Thanks @Sweetchuck. Can you add unit tests in https://github.com/pdepend/pdepend/blob/master/src/test/php/PDepend/EngineTest.php to cover you use cases?

File: src/main/php/PDepend/Engine.php [hunk 1]
```diff
@@ -635,7 +637,11 @@ private function createFileIterator()
         $fileIterator = new \AppendIterator();
 
         foreach ($this->files as $file) {
-            $fileIterator->append(new Iterator(new \GlobIterator($file), $this->fileFilter));
+            if (strpos($file, 'php://') === 0) {
+                $fileIterator->append(new \ArrayIterator(array(new \SplFileObject($file))));
+            } else {
+                $fileIterator->append(new Iterator(new \GlobIterator($file), $this->fileFilter));
+            }
```
Comment (kylekatarnls) [inline src/main/php/PDepend/Engine.php]: We could avoid the else and the duplication of `$fileIterator->append()` with a ternary.

**Comment by tvbeek** [comment]
@Sweetchuck do you have a change to add tests to this PR?

**Comment by Sweetchuck** [comment]
@tvbeek Why do you ask this?
> do you have a change to add tests to this PR?

**Comment by kylekatarnls** [comment]
@Sweetchuck we're glad you proposed this feature, but we need to cover it with unit tests. It would be more relevant to do it yourself, this way you can show exactly how you would use this feature.

**Comment by Sweetchuck** [comment]
In the original issue #361 I wrote how I would like to use.
Currently only the physical files are supported, but I would like to use PDepend with stream wrappers, such as `php://` or `data://`
https://www.php.net/manual/en/wrappers.php

**Comment by kylekatarnls** [comment]
Hi, I read your issue, but the issue will be closed and archived. We need unit tests to show concrete working examples and ensure they will continue to work on future releases.

Then there is no `data://` so I doubt this stream is supported. Unit tests would confirm it's part of the added feature or not.

We will also need a documentation so other users know this possibility exists.

We appreciate your effort but you would understand than if we merge features without tests, each change may break a previous feature without we know it (no Travis-CI warning). We try to release sustainable tested features.

So we'll merge this when someone (you, our team, or any volunteer) will propose unit tests that cover the code of this feature.

Thanks.

**Comment by Sweetchuck** [comment]
Ok, I understand, you can close the pull request, but the issue or the feature request is still open.

**Comment by kylekatarnls** [comment]
@Sweetchuck again, it's not because we don't merge your PR now, that we don't want it. We want the feature and we will merge it sooner or later. But our efforts go in priority in fixing bugs, supporting missing PHP syntax features and cover and document existing feature. Because a software fill a ton of features is useless if you users can't find how they work and can't get them working properly. The new maintenance team has a long inactivity time to catch up. After that if no one else added tests/doc here, we will take care of this.

**Comment by tvbeek** [comment]
I will not approve this before #421 is merged to prevent that we merge it to fast 😃

**Comment by kylekatarnls** [comment]
#421 merged.

**Comment by kylekatarnls** [comment]
Master branch merged to update unit tests for this PR.

**Comment by Sweetchuck** [comment]
Thank you everybody

**Pull request merged**